src/qtiCommonRenderer/renderers/interactions/TextEntryInteraction.js

```diff
@@ -134,7 +134,7 @@ var render = function render(interaction) {
 
```

Comment:
Consider adding an early return instead.
```suggestion

if (!$input.val().length) {
    return;
}

```

Reply:
I refactor because current solution don't fits with tests

Reply:
I didn't quite get you, could you please rephrase?

Reply:
OK, now I see that the build was failing. Well, of course it doesn't fail now, as the PR doesn't actually change the behavior of this method at all.
Are you sure, though, the test is correct, and doesn't require a fix as well?

Reply:
The problem is test is old and only verifies value, not pattern mask, so it's more added functionality to test than fix it. And your code, as well, don't check pattern mask. If user write some characters, length > 0 but pattern mask maybe dont match

Reply:
That's true.
At what point do we trigger the check then? Perhaps, it should be triggered on `blur` and `keyup` only.

Reply:
Its trigger every time the interaction is rendered, focus, keyup and has a pattern mask attribute.

Reply:
On blur the interaction hide the tooltip

```
.on('blur.commonRenderer', function() {
                hideTooltip($input);
            });
```